src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -0,0 +1,522 @@
+    @Override
+    public boolean isSameNodeInfo(NodeInfo nodeInfo) {
+        return this == nodeInfo;
+    }
+
+    /**
+     * Returns if implementation provides fingerprints.
+     * @return {@code boolean}
+     */
+    @Override
+    public boolean hasFingerprint() {
+        return false;
+    }
+
+    /**
+     * Returns uri of the namespace for the current node.
+     * @return uri
+     */
+    // -@cs[AbbreviationAsWordInName] Implements method from the interface.
+    // No way to change method name, because interface is the part of the Saxon library.
+    @Override
+    public String getURI() {
+        return "";
+    }
+
+    /**
+     * Returns if current node has children.
+     * @return if current node has children
+     */
+    @Override
+    public boolean hasChildNodes() {
+        return !children.isEmpty();
+    }
+
+    /**
+     * Determines axis iteration algorithm.
+     * @param axisNumber element from {@code AxisInfo}
+     * @param nodeTest filter for iterator
+     * @return {@code AxisIterator} object
+     */
+    @Override
+    public AxisIterator iterateAxis(byte axisNumber, NodeTest nodeTest) {
+        AxisIterator axisIterator = iterateAxis(axisNumber);
+        if (nodeTest != null) {
+            axisIterator = new Navigator.AxisFilter(axisIterator, nodeTest);
+        }
+        return axisIterator;
+    }
+
+    /**
+     * Determines axis iteration algorithm.
+     * @param axisNumber element from {@code AxisInfo}
+     * @return {@code AxisIterator} object
+     */
+    @Override
+    public AxisIterator iterateAxis(byte axisNumber) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns line number.
+     * @return line number
+     */
+    // -@cs[SimpleAccessorNameNotation] Implements method from the interface.
+    @Override
+    public int getLineNumber() {
+        return lineNo;
+    }
+
+    /**
+     * Returns column number.
+     * @return column number
+     */
+    // -@cs[SimpleAccessorNameNotation] Implements method from the interface.
+    @Override
+    public int getColumnNumber() {
+        return columnNo;
+    }
+
+    /**
+     * Returns parent.
+     * @return parent
+     */
+    @Override
+    public NodeInfo getParent() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns underlying node.
+     * @return underlying node
+     */
+    public DetailAST getUnderlyingNode() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns root.
+     * @return root
+     */
+    @Override
+    public NodeInfo getRoot() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns namespace array.
+     * @param namespaceBindings namespace array
+     * @return namespace array
+     */
+    // -@cs[UseVarargs] This method is the implementation of the interface method.
+    // No way to change method parameters, because interface is the part of the Saxon library.
```

Comment:
@romani 
It is an incorrect implementation of overridden method.

@timurt 
Please, use varargs as Checkstyle suggests